aeppl/censoring.py

```diff
@@ -0,0 +1,214 @@
+from typing import List, Optional
+
+import aesara.tensor as at
+import numpy as np
+from aesara.graph.basic import Node
+from aesara.graph.fg import FunctionGraph
+from aesara.graph.rewriting.basic import node_rewriter
+from aesara.scalar.basic import Ceil, Clip, Floor, RoundHalfToEven
+from aesara.scalar.basic import clip as scalar_clip
+from aesara.tensor.elemwise import Elemwise
+from aesara.tensor.var import TensorConstant
+
+from aeppl.abstract import MeasurableVariable, assign_custom_measurable_outputs
+from aeppl.logprob import CheckParameterValue, _logcdf, _logprob, logdiffexp
+from aeppl.rewriting import measurable_ir_rewrites_db
+
+
+class MeasurableClip(Elemwise):
+    """A placeholder used to specify a log-likelihood for a clipped RV sub-graph."""
+
+
+MeasurableVariable.register(MeasurableClip)
+
+
+@node_rewriter(tracks=[Elemwise])
+def find_measurable_clips(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[MeasurableClip]]:
+    # TODO: Canonicalize x[x>ub] = ub -> clip(x, x, ub)
```

Comment:
Do/will we have an issue for this?

Reply:
I'll add one

Reply:
https://github.com/aesara-devs/aeppl/issues/177